docs/development/design/build-images.rst

```diff
@@ -115,48 +125,77 @@ to install a specific Python/Conda version.
    https://github.com/readthedocs/readthedocs-docker-images/pull/166
 
 
-Specifying extra user's dependencies
-------------------------------------
-
-Different users may have different requirements. We were already requested to install
-``swig``, ``imagemagick``, ``libmysqlclient-dev``, ``lmod``, ``rust``, ``poppler-utils``, etc.
+Specifying extra language requirements
+--------------------------------------
 
-People with specific dependencies will be able to install them as APT packages or as extras
-using ``.readthedocs.yaml`` config file. Example:
+Different users may have different requirements.
+People with specific language dependencies will be able to install them by using ``.readthedocs.yaml`` config file.
+Example:
 
 .. code:: yaml
 
+   version: 3
```

Comment:
If we can easily add config constraints that flag conflicts between `build.image` and `build.os`/etc, `version: 2` seems good. This way we aren't maintaining a special configuration class or handling UX around the version number being wrong.

Reply:
Agreed, I don't think we can to deal with another version and confusion there. I think we can basically just say the 2 keys are mutually exclusive, and error if they are present together. 

I totally didn't even see the version had been incremented when I looked at the rendered version, and it will save a lot of headaches.

Reply:
OK. I will keep `version: 2` then. We can add constraints between the new configs and `build.image` 👍🏼